
'''
7.4. Написать функцию, которая принимает на вход число от 1 до 100. Если число равно 13, функция поднимает исключительную ситуации ValueError, иначе возвращает введенное число, возведенное в квадрат.
Далее написать основной код программы. Пользователь вводит число. Введенное число передаем параметром в написанную функцию и печатаем результат, который она вернула. Обработать возможность возникновения исключительной ситуации, которая поднимается внутри функции.
'''

def process_number(number):
    if number == 13:
        raise ValueError
    else:
        return number**2

while 1:
    number = int(input('Введите число от 1 до 100: '))
    if (0 < number < 101):
        break
    else:
        print('Вы ввели неверное число')

try:
    returned_number = process_number(number)
except:
    print('Ошибка: введено число 13')
else:
    print(returned_number)

# Результат:
#
# 1)
# Введите число от 1 до 100: 250
# Вы ввели неверное число
# Введите число от 1 до 100: 25
# 625
#
# 2)
# Введите число: 13
# Ошибка: введено число 13
